lib/v2/screens/sign_up/create_username/mappers/create_username_mapper.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/states/create_username_state.dart';
+
+class CreateUsernameMapper extends StateMapper {
+  SignupState mapValidateUsernameToState(SignupState currentState, Result result) {
+    final createUsernameCurrentState = currentState.createUsernameState;
+
+    if (result.isError) {
+      // Error means username is not taken and is available for the user to take it
```

Comment:
this is so weird... Is this a chain limitation? 
is this how its done in v1?

Reply:
Yup 🤷🏽‍♂️